### ISSUE TITLE
Why there is a strict `rake '~>13.0'` requirement in gemspec?

### ISSUE THREAD

**Issue by manuyavuz-pointr** [postmodern/digest-crc#24]
Hello,

I would like to ask why there is a strict version requirement in gemspec of `0.6.1` version.

It currently breaks existing systems where the original `Gemfile` does not have a specific `rake` dependency defined.

Default `rake` that exists in ruby installations has version 12 for ruby versions 2.6 and a few earlier ones. With `digest-crc` gemspec change, calling `rake` directly has started to fail saying we should run with `bundle exec rake`.

In short, is it possible to have a wider range of supported rake versions there?

Thanks

**Comment by postmodern** [comment]
I can lower the version requirement to `~> 12.0`.

**Comment by manuyavuz-pointr** [comment]
@postmodern sorry I have seen your reply now.

Wouldn't this then break for ruby 2.7 which has rake 13 by default?

**Comment by postmodern** [comment]
@manuyavuz-pointr Oops, you are correct. How about 06380bd?

**Comment by postmodern** [comment]
@manuyavuz-pointr oops, you're right. I broadened it to include from `>= 12.0.0` up to `< 14.0.0`.
https://github.com/postmodern/digest-crc/blob/577178fce57c2a5b14a3d74bb0e590e73a7574a6/gemspec.yml#L25

**Comment by v-kolesnikov** [comment]
I have the same issue. It was quite unexpected to see that this gem directly depends from Rake. It is a quite rarely case for gems. @postmodern Could you explain shortly why digest-crc depends from Rake? Thanks!

**Comment by v-kolesnikov** [comment]
I've check the code and issues. Looks like Rake is required to build native extensions for this gem, rigth?

**Comment by postmodern** [comment]
See #18. Yep bundler will exclude rake, even though it's a built-in gem of Ruby. Rake is used to compile the C extensions, but rescue any exceptions if the C extensions fail to compile for whatever reason; this allows digest-crc to successfully install even if the C extensions fail to compile, and digest-crc fallback to using the pure-Ruby versions of the `#update` methods.

Is a broad version requirement of `>= 12.0.0, < 14.0.0` sufficient?

**Comment by postmodern** [comment]
RubyGems does also support building C extensions using a [`./configure` shell script and a `Makefile`](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/ext/configure_builder.rb). However, this method probably would not work on Windows...

**Comment by v-kolesnikov** [comment]
> Is a broad version requirement of >= 12.0.0, < 14.0.0 sufficient?

I think it's good enough. But version with that fix (06380bd) is not released yet. Could you make a release please?
Many thanks!

**Comment by postmodern** [comment]
Released 0.8.3.
